src/graphics/sdl/sdl_graphics.cpp

```
@@ -227,7 +264,43 @@ namespace moth_graphics::graphics::sdl {
+        if (rotation == 0.0) {
+            // Fast path: no rotation — draw directly at the transformed world position.
+            auto const worldPos = t.TransformPoint({ static_cast<float>(destRect.topLeft.x) + x,
+                                                     static_cast<float>(destRect.topLeft.y) + y });
+            FC_DrawColumnEffect(fcFont.get(), m_surfaceContext.GetRenderer(), worldPos.x, worldPos.y, destWidth, effect, "%s", text.c_str()); // NOLINT(cppcoreguidelines-pro-type-vararg)
+        } else {
+            // Rotation path: render text to a temp texture, then draw that texture rotated.
+            auto tempTex = CreateTextureRef(SDL_CreateTexture(m_surfaceContext.GetRenderer(),
+                SDL_PIXELFORMAT_RGBA8888, SDL_TEXTUREACCESS_TARGET, destWidth, destHeight));
+            SDL_SetTextureBlendMode(tempTex->GetImpl(), SDL_BLENDMODE_BLEND);
+
+            SDL_Texture* prevTarget = SDL_GetRenderTarget(m_surfaceContext.GetRenderer());
```

Clang (14.0.6)

[error] variable 'prevTarget' is not initialized

(cppcoreguidelines-init-variables,-warnings-as-errors)